overlord/snapstate/snapstate.go

```diff
@@ -153,6 +167,17 @@ func doInstall(s *state.State, snapst *SnapState, ss *SnapSetup) (*state.TaskSet
 			prev = tasks[len(tasks)-1]
 		}
 
+		// discard the first As in A-...-A refreshes
+		for i := 0; i < currentIndex; i++ {
+			si := seq[i]
+			if si.Revision == ss.Revision() {
+				// we do *not* want to removeInactiveRevision of this one
+				seq = seq[:i+copy(seq[i:], seq[i+1:])]
```

Comment:
I agree! I thought this weirdly packed construction was idiomatic (given it's in the slice tricks doc).